sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -2031,4 +2031,13 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
     new SparkException(errorClass = "NULL_COMPARISON_RESULT",
       messageParameters = Array(), cause = null)
   }
+
+  def invalidPatternError(funcName: String, pattern: String): RuntimeException = {
+    new SparkRuntimeException(
+      errorClass = "INVALID_PARAMETER_VALUE",
+      messageParameters = Array(
+        "regexp",
+        toSQLId(funcName),
+        pattern))
```

Comment:
should we quote the pattern? It's a string value.

Reply:
Not sure, the pattern can include anything. I would not add extra symbols to it, and show it to users as is.

Reply:
how do we show string value in the error message? string value can include anything as well.

Reply:
String values should be quoted by ''

Reply:
Then what's the difference here? Why can't we quote the regex string?

Reply:
ok. Let me create a task w/ tech debts related to errors/error classes.

Reply:
I think we can display string `a'b` as `'a\'b'`

Reply:
It's an existing issue and I don't think it should stop us from quoting regex strings.

Reply:
I opened JIRA to address tech debts like this: https://issues.apache.org/jira/browse/SPARK-39778